src/audiences/edit/data/index.js

```diff
@@ -57,128 +64,66 @@ const actions = {
-					...state,
-					posts,
-				};
-			}
-			case 'SET_POST': {
-				let posts = state.posts.slice();
-				if ( action.post.id ) {
-					posts = posts.map( post => {
-						if ( post.id === action.post.id ) {
-							post = {
-								...post,
-								...action.post,
-							};
-						}
-						return post;
-					} );
-				}
-				return {
-					...state,
-					posts: action.post.id ? posts : state.posts,
-					post: {
-						...state.post,
-						...action.post,
-					},
-				};
-			}
-			default: {
-				return state;
-			}
-		}
+const selectors = {
+	getFields( state ) {
+		return state.fields;
 	},
-	actions,
-	selectors: {
-		getFields( state ) {
-			return state.fields;
-		},
-		getEstimate( state, audience ) {
-			const key = JSON.stringify( audience );
-			return state.estimates[ key ] || {
-				count: 0,
-				total: 0,
-				histogram: new Array( 28 ).fill( { count: 1 } ), // Build empty histrogram data.
-			};
-		},
-		getPost( state ) {
-			return state.post;
-		},
-		getPosts( state ) {
-			return state.posts;
-		},
+	getEstimate( state, audience ) {
+		const key = JSON.stringify( audience );
+		return state.estimates[ key ] || {
+			count: 0,
+			total: 0,
+			histogram: new Array( 28 ).fill( { count: 1 } ), // Build empty histrogram data.
+		};
 	},
-	controls: {
-		FETCH_FROM_API( action ) {
-			return apiFetch( action.options );
-		},
+	getPost( state ) {
+		return state.post;
 	},
-	resolvers: {
-		* getFields() {
-			const fields = yield actions.fetch( {
-				path: 'analytics/v1/audiences/fields',
-			} );
-			return actions.setFields( fields );
-		},
-		* getEstimate( audience ) {
-			const audienceQuery = encodeURIComponent( JSON.stringify( audience ) );
-			const estimate = yield actions.fetch( {
-				path: `analytics/v1/audiences/estimate?audience=${ audienceQuery }`,
-			} );
-			return actions.addEstimate( audience, estimate );
-		},
-		* getPost( id ) {
-			const post = yield actions.fetch( {
-				path: `wp/v2/audiences/${ id }?context=edit`,
-			} );
-			if ( post.status === 'auto-draft' ) {
-				post.title.rendered = '';
-			}
-			if ( ! post.audience ) {
-				post.audience = defaultAudience;
-			}
-			yield actions.addPosts( [ post ] );
-			return actions.setPost( post );
-		},
-		* getPosts( page = 1, search = '' ) {
-			const posts = yield actions.fetch( {
-				path: `wp/v2/audiences?context=edit&page=${ page }&search=${ search }`,
-			} );
-			return actions.addPosts( posts );
-		},
+	getPosts( state ) {
+		return state.posts;
 	},
+};
+
+const resolvers = {
+	*getFields() {
+		const fields = yield actions.fetch( {
+			path: 'analytics/v1/audiences/fields',
+		} );
+		return actions.setFields( fields );
+	},
+	*getEstimate( audience ) {
+		const audienceQuery = encodeURIComponent( JSON.stringify( audience ) );
+		const estimate = yield actions.fetch( {
+			path: `analytics/v1/audiences/estimate?audience=${ audienceQuery }`,
+		} );
+		return actions.addEstimate( audience, estimate );
+	},
+	*getPost( id ) {
```

Comment:
Where is this used? I'm looking to understand if there's really value to using generators here as they are seldom used, especially by us.

Reply:
I believe this is used so we can essentially return multiple values from the function; not sure though

Reply:
Mmm ok I guess we can leave that to the parent PR